builtin/providers/infoblox/resource_infoblox_record.go

```diff
@@ -0,0 +1,258 @@
+package infoblox
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/fanatic/go-infoblox"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceInfobloxRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceInfobloxRecordCreate,
+		Read:   resourceInfobloxRecordRead,
+		Update: resourceInfobloxRecordUpdate,
+		Delete: resourceInfobloxRecordDelete,
+
+		Schema: map[string]*schema.Schema{
+			"domain": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"value": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"ttl": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "3600",
+			},
+		},
+	}
+}
+
+func resourceInfobloxRecordCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*infoblox.Client)
+
+	record := url.Values{}
+	if err := getAll(d, record); err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Infoblox Record create configuration: %#v", record)
+
+	var recId string
+	var err error
+
+	switch strings.ToUpper(d.Get("type").(string)) {
+	case "A":
+		opts := &infoblox.Options{
+			ReturnFields: []string{"ttl", "ipv4addr", "name"},
+		}
+		recId, err = client.RecordA().Create(record, opts, nil)
+	case "AAAA":
+		opts := &infoblox.Options{
+			ReturnFields: []string{"ttl", "ipv6addr", "name"},
+		}
+		recId, err = client.RecordAAAA().Create(record, opts, nil)
+	case "CNAME":
+		opts := &infoblox.Options{
+			ReturnFields: []string{"ttl", "canonical", "name"},
+		}
+		recId, err = client.RecordCname().Create(record, opts, nil)
+	}
```

Comment:
Should we have a `default` case here that returns an error? Otherwise I think this will fail strangely when the type is not what's expected, because the `recId` will be the empty string when we get to `d.SetId` below.